optuna/storages/rdb/storage.py

```diff
@@ -848,6 +850,24 @@ def _merge_trials_orm(
 
         return result
 
+    @staticmethod
+    def _set_default_engine_kwargs_for_mysql(url, engine_kwargs):
+        # type: (str, Dict[str, Any]) -> None
+
+        # Skip if RDB is not MySQL.
+        if not url.startswith('mysql'):
+            return
+
+        # Do not overwrite value.
+        if 'pool_pre_ping' in engine_kwargs:
+            return
+
+        # If True, the connection pool checks liveness of connections at every checkout.
+        # Without this option, trials that take longer than `wait_timeout` may cause connection
+        # errors. For further details, please refer to the following document:
+        # https://docs.sqlalchemy.org/en/13/core/pooling.html#pool-disconnects-pessimistic
+        engine_kwargs['pool_pre_ping'] = True
```

Comment:
Are these engine options presented to the user in some way? If not, the `RDBStorage` holds a logger, how about logging this internal behavior with `logger.debug` as it this options will affect users but isn't necessarily easy to spot? I'm also wondering where this should be documented in `__init__` as the `engine_kwargs` is part of the public API.

Reply:
> Are these engine options presented to the user in some way?

No, they aren't. As you mentioned, it will be hard for users to know the actual options given to MySQL client.  I'll add a log message for debugging.

> this should be documented in __init__ as the engine_kwargs is part of the public API.

Thank you for your suggestion. You're right.
It should also be mentioned in the document. I'll fix it.

Reply:
I added the log message and the document about `pool_pre_ping`. I have two comments about the change.

1) Log message

I cannot use `self.logger` because `_set_default_engine_kwargs_for_mysql` is a static method. `_commit_with_integrity_check` also instanciates the logger, and I think we can change the logger to a module variable (c.f., https://github.com/optuna/optuna/pull/770).

2) Document

I added the description as a note instead of a part of `engine_kwargs` because it is not related to all users of `RDBStorage`. What do you think of it?

![image](https://user-images.githubusercontent.com/3255979/73160499-85f4c980-412c-11ea-9ca1-798fbbdd2dd7.png)

Reply:
> 1. Log message

Ah, I see. Then what do you say about addressing that in a separate PR?

> 2. Document

Thanks, looks great. I left some minor comments though if you could take a look.

Reply:
>> Log message
> Ah, I see. Then what do you say about addressing that in a separate PR?

Sounds good.

>> Document
> Thanks, looks great. I left some minor comments though if you could take a look.

Sorry for my careless mistake. I fixed them.